### PR TITLE
Add wrap utility helper documentation

### PR DESCRIPTION
**Pull request by dentuzhik**
See https://github.com/airbnb/enzyme/issues/919 for my monolog of finding this in source code :)
I know explanation in documentation may not be the best, but it's better then nothing.

### CODE REVIEW

File: docs/api/ShallowWrapper/wrap.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+
+
+#### Examples
+
+```jsx
+const wrapper = shallow(<MyComponent />);
+const myOtherWrapper = wrapper.wrap(<MyOtherComponent />);
+
+expect(myOtherWrapper.instance()).to.be.instanceOf(MyOtherComponent);
```
Comment (ljharb) [inline docs/api/ShallowWrapper/wrap.md]: this could probably be explained a bit better by providing `.debug()` output for both wrappers?

Comment (dentuzhik) [inline docs/api/ShallowWrapper/wrap.md]: Fair enough, I'll update it tomorrow.

**Comment by xaka** [comment]
@dentuzhik `wrap` is we've been looking for! thanks for discovering it :) 

it's been a while, but do you think you'll have time to address that minor PR comment, rebase and get it merged? thanks again

**Comment by lelandrichardson** [comment]
@xaka @dentuzhik i'm curious... what is it you're using `wrap` for?  It's traditionally been thought of as an internal method and was thus not documented. I'm a bit reluctant to make it part of our public API, but am curious to know why you found it useful?

**Comment by dentuzhik** [comment]
I have described my usage pattern in the related issue, basically it makes it possible to test chunks of html extracted from a main render method to other component methods (like "renderButton").

I mostly use this scenario when I am not entirely sure that this chunk of markup is reusable, but don't want to leave it uncovered. This way I have a method, which is almost prepared to extraction to the component with an isolated test, which will not fail if the tests for the main render will.

I am willing to rebase PR and provide addition description as requested.

**Comment by dentuzhik** [comment]
After a few years of keeping this PR opened, for those who will come by I will say that I agree with original reasoning of @ljharb regarding SFCs.

Especially after having changed context to TS, where type safety of SFC is way more straightforward than for component methods which return React elements.

So @lelandrichardson I think it shouldn't be in public API :)

**Pull request closed without merging**